### DISCUSSION TITLE
Disk space estimation for locally stored content files

### DISCUSSION THREAD

**Discussion by novice002** [WikiTransformationProject/wikitraccs-releases#105] [Q&A] [answered]
Hello,

We have quite number of Confluence spaces to migrate, 
and I am concerned on measuring the needed disk spaces for the extracted Confluence files (to be stored locally).

Is there any calculation so that I don't have to worry about the storage running out of spaces in the middle of migration?
(2x times the total content file size, for example)

Thank you

**Comment by heinrich-ulbricht** [comment, marked as answer]
@novice002

WikiTraccs by default uses the logged-in user's profile to store attachments downloaded from Confluence. You can change this location. Ultimately at the chosen location there must be enough free space available to store the attachments of a migration run. Attachments will be downloaded for each migrated page, and pages are migrated one after another, so the space needed rises gradually. WikiTraccs checks if there is enough space available and will pause a running migration if available disk space is below a certain threshold (~2 GB). This will be shown in the console with a corresponding message. When there is enough space available again it will continue.

You can control in which directory WikiTraccs stores downloaded attachments during the migration to SharePoint, using **appsettings.json**. See this article for a sample configuration: [Change temporary storage folder pathes](https://www.wikitransformationproject.com/docs/reference/settings/configuration-file/settings-samples/#snippet-change-temporary-storage-folder-pathes).

WikiTraccs won't clean up those files by itself, but you can delete them at any time. If files are missing in this directory they will be downloaded again.

With regard to **storage requirements** - Atlassian shows how to calculate the size of attachments: [How to find the total size of all attachments in Confluence](https://confluence.atlassian.com/confkb/how-to-find-the-total-size-of-all-attachments-in-confluence-794201234.html). The approach taken is to look at the Confluence database, and this is also an often used approach I see in migration projects.

Let me know if you need more information.

**Reply by novice002** [reply]
Thank you so much for your insight and details - Much appreciated!!

Among the given all valuable information, the following was one the things I wanted to check and you confirmed: 
'_...you can delete them at any time. If files are missing in this directory they will be downloaded again._'

While we are at it, can I ask you one more question?
What happens if I perform migration multiple times from the identical Confluence source space to the identical target SPO site?

For example, Space 'SPC1' was migrated to SPO site 'SPO1'. 
Then perform the migration on the identical source/target in after a few days.

I was hoping that doing so would act as 'delta' migration. But I wanted to get your expertise on this situation.

I am _not_ really seeking complicated 'delta' migration functions (like WikiTraccs calculating/pinpointing discrepancies). 
As long as the existing contents get overwritten by newer migrated contents, that would work for me :)

**Reply by heinrich-ulbricht** [reply]
WikiTraccs does not overwrite by default. You could delete the SharePoint pages before migrating again. Or you can use the progress log files to explicitly mark pages to overwrite: https://www.wikitransformationproject.com/docs/reference/update-old-pages/#how-to-tell-wikitraccs-to-update-outdated-pages

Seems like a simple "just overwrite, I don't care about existing content" button is missing 😅

**Comment by novice002** [comment]
Thank you again for sharing your knowledge.
Those approaches would work for me.

In my humble opinion regarding that _simple_ button, it never feels 'simple' to me - as it could mean updates on the existing UI and logic  
:-o
